specification/under_development/Instruction_Set.md

```diff
@@ -0,0 +1,52 @@
+To target a program to a specific backend hence requires selecting both a profiles and QIS that is supported by that backend, see also [Compilation and Targeting](Compilation_And_Targeting.md). 
+
+## Runtime Functions
+
+Instructions that execute on a classical processor or serve data transfer are expressed as LLVM instructions, including calls to `__quantum__rt__*` functions.
+These functions are forward declared in the IR and defined by the executing runtime. Which LLVM instructions and runtime functions are used/supported is captured in the QIR profile specification. The complete list of all runtime functions that are needed for executing QIR programs independent on which profile they have been compiled for is given in the [Library_Reference.md](). Most backends do not support the full set of runtime functions but merely a subset that is sufficient for the supported QIR profile(s). The [Base Profile](profiles/Base_Profile.md) defines the minimal set of runtime functions that need to be supported for by a backend to permit computations that make use of qubits and measurements of qubits.Additional functions may be needed to support the use of data types beyond those for qubit and measurement result values.
+
+## Quantum Instruction Set (QIS)
+
+The QIR specification contains a [table](../QuantumInstructions.md) of commonly used instructions along with their signatures and a description of their functionality.
+Backends are **not** required to support all of these. Instead, each backend will declare which of these instructions it supports. We encourage to make instructions that are supported by a context independent implementation in terms of other instructions as a library rather than listing them as part of the backend specification. A library provided either in the form of a bitcode file can be linked in as part of a QIR compilation stage. 
+
+
+All quantum instructions are represented by LLVM external functions.
+Quantum instructions may take qubits, doubles, or integers as parameters,
+and should all have no return; that is, they should be void.
+
+Measurements should take an offset into the `ClassicalStorage` global as a parameter.
+The measurement result should be stored into the corresponding bit in `ClassicalStorage`.
+
+The LLVM functions that implement the quantum instruction set should all have
+names that start with `__quantum__qis__`.
+
+QIR does not specify the contents of the quantum instruction set.
+However, in order to ensure some amount of uniformity, implementations that provide
+any of the following quantum instructions must match the specified definition:
```

Comment:
Do we want to pin down phase in the gate definitions? Although it makes no observable difference to the operations themselves, it becomes relevant when one wants to specify controlled versions of them, for example.
I would suggest doing so, as there is no harm in it and it could be useful for future extensions.

Reply:
Agree, I think phases need to be specified, since they matter for a controlled version, and compilers may rely on the relation between a controlled version and the gate (so controlled version are not spec'ed independently).